### PR TITLE
Prime: Implement refills to the item pool.

### PR DESCRIPTION
**Pull request by MrMiguel211**
image
GM8E01_2024-03-13_15-35-09

### CODE REVIEW

**Comment by Miepee** [comment]
Needs fixed tests, I'll do those later

**Comment by Miepee** [comment]
New screenshots after the long name rename:
GUI:
grafik

In-Game:
grafik
grafik
grafik

File: randovania/games/prime1/pickup_database/pickup-database.json
Comment (toasterparty) [inline randovania/games/prime1/pickup_database/pickup-database.json:550]: `Health` -> `Energy` for player-facing text

File: randovania/games/prime1/pickup_database/pickup-database.json
Comment (toasterparty) [inline randovania/games/prime1/pickup_database/pickup-database.json:571]: Replace `a set amount`  by how much they actually refill

**Review by toasterparty — CHANGES_REQUESTED** [review]

**Comment by toasterparty** [comment]
Did this really just work in MultiWorld without having to finagle anything?

**Comment by Miepee** [comment]
MW wasnt tested yet. Wanted to do this in beta builds because its easier to test.

**Comment by henriquegemignani** [comment]
> Did this really just work in MultiWorld without having to finagle anything?

Does giving the health work by incrementing the amount/capacity of an given item or is there any extra functionality?

**Comment by henriquegemignani** [comment]
Oh there's also missile and pb refills. I'm pretty confident these will need special handling to receive, otherwise they'll also give capacity.

**Review by toasterparty — APPROVED** [review]
I'm approving even though I have doubts about this actually working first try. Please test as soon as this hits the server and don't merge unless you're going to be able to test that day, thanks.

**Comment by henriquegemignani** [comment]
> I'm approving even though I have doubts about this actually working first try. Please test as soon as this hits the server and don't merge unless you're going to be able to test that day, thanks.

Swap my "pretty confident" to 100% sure. Receiving a missile or pb refill remotely will give you capacity. Dunno about energy refill though.

**Review by toasterparty — CHANGES_REQUESTED** [review]
un-approving, please demonstrate that this is working on a bench multiworld

**Comment by toasterparty** [comment]
tests? conflict.

**Comment by Miepee** [comment]
tests are normal, we're a bit flakey
conlfict is expected. I couldnt figure out yet how to make this work with MW. 
In the meantime, I figured out tho how to mark presets with refill items as incompatible for MW, would that be an acceptible workaround to get this merged?

**Comment by Miepee** [comment]
I just remembered this discusssion:
grafik


I could try pursuing this further, but this wont happen before next month. So up to you if this should get longer in limbo, or if we just exclude them from MW. I'm fine with either.

**Comment by ThanatosGit** [comment]
What's the status here?
If it isn't mergable and nobody is working on that, I prefer to change it to a Draft.

**Comment by Miepee** [comment]
Status is "needs MW functionality".
Draft sounds like a good idea.

**Comment by Miepee** [comment]
@toasterparty this should be rebased, and Jeff told in prime-dev that it worked in MW now.
Hopefully we can get this merged soon :)

**Review by toasterparty — CHANGES_REQUESTED** [review]
- Fix merge conflict
- Add refill pickups to one of the test rdvgame files

**Comment by Miepee** [comment]
Fixed/Added both.

**Review by toasterparty — APPROVED** [review]

**Pull request merged**
